sklearn/linear_model/_ridge.py

```diff
@@ -795,22 +806,72 @@ def _ridge_regression(
         )
 
     if solver == "svd":
-        if sparse.issparse(X):
+        if X_is_sparse:
             raise TypeError("SVD solver does not support sparse inputs currently")
-        coef = _solve_svd(X, y, alpha)
+        coef = _solve_svd(X, y, alpha, xp)
 
     if ravel:
-        # When y was passed as a 1d-array, we flatten the coefficients.
-        coef = coef.ravel()
+        coef = _ravel(coef)
+
+    coef = xp.asarray(coef)
 
     if return_n_iter and return_intercept:
-        return coef, n_iter, intercept
+        res = coef, n_iter, intercept
     elif return_intercept:
-        return coef, intercept
+        res = coef, intercept
     elif return_n_iter:
-        return coef, n_iter
+        res = coef, n_iter
     else:
-        return coef
+        res = coef
+
+    return (*res, solver) if return_solver else res
+
+
+def _solver_auto_set(solver, positive, return_intercept, is_sparse, xp):
+    if solver != "auto":
+        return solver
+
+    is_numpy_namespace = _is_numpy_namespace(xp)
+
+    auto_solver_np = _solver_auto_set_np(positive, return_intercept, is_sparse)
+    if is_numpy_namespace:
+        return auto_solver_np
+
+    if positive:
+        raise ValueError(
+            "The solvers that support positive fitting do not support "
+            f"Array API dispatch to namespace {xp.__name__}. Please "
+            "either disable Array API dispatch, or use a numpy-like "
+            "namespace, or set `positive=False`."
+        )
+
+    # At the moment, Array API dispatch only supports the "svd" solver.
+    solver = "svd"
+    if solver != auto_solver_np:
+        warnings.warn(
+            f"Using Array API dispatch to namespace {xp.__name__} with "
+            f"`solver='auto'`will result in using the solver '{solver}'. "
+            "Results might be different than when Array API dispatch is "
```

Comment:
Some suggestions for rewording:
* The results may differ from using Numpy or disabling Array API dispatch, because in that case the preferred solver would be {solver}.
* Using a Numpy array or disabling Array API dispatch might produce different results because in that case the preferred solver would be {solver}.
* The results may differ from those when using a Numpy array, because in that case the preferred solver would be {solver}.

The last suggestion is because I'm thinking that "use Numpy array" and "disable Array API" are equivalent. Not sure if this is true? If you disable Array API support, the only option is to use a Numpy array no? Or do we have the case that "Numpy + array api enabled" != "Numpy + array api disabled"?

Reply:
Done in a02e7a4.